### PR TITLE
Context diagram checksums

### PR DESCRIPTION
**Pull request by micha91**
A complete rework of checksums for attachments
requires [this PR](https://github.com/DSD-DBS/capellambse-context-diagrams/pull/147) to be merged first

### CODE REVIEW

File: capella2polarion/data_model/work_item_attachments.py
Comment (ewuerger) [inline capella2polarion/data_model/work_item_attachments.py:75]: Output isn't typed. 
```suggestion
    def content_bytes(self) -> bytes:
```

File: capella2polarion/data_model/work_item_attachments.py
Comment (ewuerger) [inline capella2polarion/data_model/work_item_attachments.py:113]:
```suggestion
    def content_checksum(self) -> str:
```

File: capella2polarion/data_model/work_item_attachments.py
Comment (ewuerger) [inline capella2polarion/data_model/work_item_attachments.py:159]:
```suggestion
    def content_bytes(self) -> bytes:
```

File: capella2polarion/data_model/work_items.py
Comment (ewuerger) [inline capella2polarion/data_model/work_items.py:58]: Please type it, bonus points for inventing a Type-Alias.

**Review by ewuerger — CHANGES_REQUESTED** [review]
Very nice and clean way to check if attachments need an update or not.
There are some minor things that I recognized, mostly about untyped outputs.

A question:
We are using bytes only here and as they are immutable, on any operation of it a new bytes object is created and stored in memory. How often do we actually read images from their path to receive the bytes? With `io.BytesIO` there is an improvement in speed and memory allocation (can be neglected I think) for the read operations. But if we don't do that so often there is no performance gain here, switching from just handling bytes to handling an `io.BytesIO` object.

After solving the requests, you'll get my approval.

**Comment by micha91** [comment]
> A question: We are using bytes only here and as they are immutable, on any operation of it a new bytes object is created and stored in memory. How often do we actually read images from their path to receive the bytes? With `io.BytesIO` there is an improvement in speed and memory allocation (can be neglected I think) for the read operations. But if we don't do that so often there is no performance gain here, switching from just handling bytes to handling an `io.BytesIO` object.

We only read files from disk when we create the attachment. The changes with the dynamic generation of the content only affect diagrams and in this case we also cache the result in `_content_bytes` so that also diagrams are only generated/loaded from disk once.

**Review by ewuerger — APPROVED** [review]
LGTM.

**Pull request merged**
